core/src/main/java/com/graphhopper/routing/ch/PrepareEncoder.java

```diff
@@ -42,20 +42,27 @@ public static final long getScBwdDir() {
     }
 
     /**
-     * Returns true if flags1 can be overwritten in the edge by flags2 without restricting or
-     * changing the directions of flags1.
-     * <p>
+     * Returns 1 if existingEdgeFlags can be overwritten in the edge by newEdgeFlags without limiting or
+     * changing the directions of existingEdgeFlags. It returns 2 for the same condition AND if the edge
+     * with newEdgeFlags has to be added even if weight is higher than existing edge weight.
+     * <pre>
+     *                    | newEdgeFlags:
+     * existingEdgeFlags  | -> | <- | <->
+     * ->                 |  1 | 0  | 2
+     * <-                 |  0 | 1  | 2
+     * <->                |  0 | 0  | 1
```

Comment:
Interesting that these two cases (with '2' as response) are that raw in reality for shortcuts so that this bug remained so long in the code.

BTW: always **adding** the shortcuts without any updates dramatically slows down the preparation time. So this tuning with canBeOverwritten/getMergeStatus is very important.